pkg/image/controller/scheduled_image_controller.go

```diff
@@ -180,6 +180,13 @@ func resetScheduledTags(stream *imagev1.ImageStream) {
 	next := stream.Generation + 1
 	for tag, tagRef := range stream.Spec.Tags {
 		if tagImportable(tagRef) && tagRef.ImportPolicy.Scheduled {
+			if ref, err := imageref.Parse(tagRef.From.Name); err != nil && len(tagRef.From.Name) > 0 {
+				continue
+			} else if len(ref.ID) > 0 && !tagNeedsImport(stream, tagRef, true) {
+				// ref.ID is set if this is a canonical, sha/digest ref;
+				// we allow scheduled import of those, but only until the initially succeed
+				continue
+			}
```

Comment:
i think it is still reasonable to also do this check in needsScheduling.  (if no tags in the imagestream end up needing scheduling, we save ourselves some work)

Reply:
ok cool yeah that is why I left it there

Reply:
oh, i must have only been looking at the new commit, so i thought you moved it to here only.  nm.